src/libsystemd-network/lldp-neighbor.c

```diff
@@ -84,6 +86,15 @@ _public_ sd_lldp_neighbor *sd_lldp_neighbor_unref(sd_lldp_neighbor *n) {
         return NULL;
 }
 
+static void lldp_neighbor_destroy(sd_lldp_neighbor *n) {
+        assert(n);
+        assert(n->lldp);
+        assert(n->n_ref <= 0);
+
+        assert_se(prioq_remove(n->lldp->neighbor_by_expiry, n, &n->prioq_idx) >= 0);
+        lldp_neighbor_free(n);
```

Comment:
Hmm, should this be `sd_lldp_neighbor_unref()`? Otherwise, if `sd_lldp_neighbor` objects are referenced and `sd_lldp` is freed, then `sd_lldp_neighbor` objects are also freed even if they are referenced.

Reply:
So I added `assert(n->n_ref <= 0);` here, in the assumption that `sd_lldp_stop()` will only be called once all references to all sd_lldp_neighbors will have been unref'd already...

(Is that really a valid assumption? The `assert()` will validate it, but potentially with a crash...)

(Calling `sd_lldp_neighbor_unref()` doesn't work, since typically n_ref is zero here, so the ref/unref logic is meant for all references *outside* the one from the LLDP itself.)

Not really sure what you mean by your other comment ("reference counter of `sd_lldp_neighbor` objects seems to be borked, ...), can you please expand on that?

Reply:
Hmm? Sorry, I may misunderstand your point or code, but I guess with your PR the following code may not work, as the ref counters of neighbors are not 1 or 0.
```
n = sd_lldp_get_neighbors(lldp, &neighbors);
(void) sd_lldp_stop(lldp);
for (i = 0; i < n; i++)
        sd_lldp_neighbor_unref(neighbors[i]);
free(neighbors);
```